### PR TITLE
Scaffolder: fix auth credentials issue

### LINKED ISSUE #23787: 🐛 Bug Report: Scaffolder catalog:fetch and catalog:register cannot access secrets
### 📜 Description

When using the scaffolder actions catalog:register and catalog:fetch the following error is displayed:

`
12024-03-22T09:20:51.276Z Beginning step Fetch Entity
22024-03-22T09:20:51.285Z TypeError: Cannot read properties of undefined (reading 'secrets')
3    at Object.getInitiatorCredentials (/app/node_modules/@backstage/plugin-scaffolder-backend/dist/cjs/router-77345506.cjs.js:2001:19)
4    at Object.handler (/app/node_modules/@backstage/plugin-scaffolder-backend/dist/cjs/router-77345506.cjs.js:364:31)
5    at NunjucksWorkflowRunner.executeStep (/app/node_modules/@backstage/plugin-scaffolder-backend/dist/cjs/router-77345506.cjs.js:2626:22)6    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)7    at async NunjucksWorkflowRunner.execute (/app/node_modules/@backstage/plugin-scaffolder-backend/dist/cjs/router-77345506.cjs.js:2729:9)8    at async TaskWorker.runOneTask (/app/node_modules/@backstage/plugin-scaffolder-backend/dist/cjs/router-77345506.cjs.js:2974:26)9    at async run (/app/node_modules/p-queue/dist/index.js:163:29)` 

This did not happen in the 1.23 version of backstage but is now in version 1.24.

### 👍 Expected behavior

The action should run as aspected.

### 👎 Actual Behavior with Screenshots

The scaffolder stops and displays the error like so:

image

### 👟 Reproduction steps

The template step I was running is the following but happens on every entity ref:

```yaml
    - id: fetchentity
      name: Fetch Entity
      action: catalog:fetch
      input:
        entityRef: ${{ parameters.tenant }}
```

The error occurrs immediately after the actions starts.

### 📃 Provide the context for the Bug.

After digging around in the code, the location where the error gets thrown is here https://github.com/backstage/backstage/blob/ed10cea9e1ae701f1ddecba1852515fba1e5edc3/plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts#L182

`this.task` is undefined in our case thus throwing the error. 

The call to this function is here https://github.com/backstage/backstage/blob/ed10cea9e1ae701f1ddecba1852515fba1e5edc3/plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetch.ts#L94.

The problem is that the ctx does not have a task field thus having an undefined object.
This can be fixed with small adjustments to the code like this:

```ts
    if (this.secrets && "initiatorCredentials" in this.secrets) {
      return JSON.parse(this.secrets.initiatorCredentials);
    }
```

But I'm not sure if this is the correct implementation. Because removing the task und renaming `__initiatorCredentials` to `initiatorCredentials` felt not right

### 🖥️ Your Environment

OS:   Darwin 23.4.0 - darwin/arm64
node: v18.17.1
yarn: 3.7.0
cli:  0.26.0 (installed)
backstage:  1.24.0

Dependencies:
  @backstage/app-defaults                                          1.5.1
  @backstage/backend-app-api                                       0.5.14, 0.6.0
  @backstage/backend-common                                        0.19.10, 0.21.4
  @backstage/backend-defaults                                      0.2.14
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.7
  @backstage/backend-plugin-api                                    0.6.14
  @backstage/backend-tasks                                         0.5.19
  @backstage/backend-test-utils                                    0.3.4
  @backstage/catalog-client                                        1.6.1
  @backstage/catalog-model                                         1.4.5
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.4
  @backstage/cli                                                   0.26.0
  @backstage/config-loader                                         1.7.0
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.12.1
  @backstage/core-compat-api                                       0.2.1
  @backstage/core-components                                       0.13.10, 0.14.1
  @backstage/core-plugin-api                                       1.9.1
  @backstage/dev-utils                                             1.0.28
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.6
  @backstage/frontend-plugin-api                                   0.6.1
  @backstage/integration-aws-node                                  0.1.10
  @backstage/integration-react                                     1.1.25
  @backstage/integration                                           1.9.1
  @backstage/plugin-api-docs                                       0.11.1
  @backstage/plugin-app-backend                                    0.3.62
  @backstage/plugin-app-node                                       0.1.14
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.6
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.5
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.9
  @backstage/plugin-auth-backend-module-github-provider            0.1.11
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.11
  @backstage/plugin-auth-backend-module-google-provider            0.1.11
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.9
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.11
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.7
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.4
  @backstage/plugin-auth-backend-module-okta-provider              0.0.7
  @backstage/plugin-auth-backend                                   0.22.0
  @backstage/plugin-auth-node                                      0.4.9
  @backstage/plugin-auth-react                                     0.0.1
  @backstage/plugin-catalog-backend-module-gitlab                  0.3.11
  @backstage/plugin-catalog-backend-module-openapi                 0.1.31
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.11
  @backstage/plugin-catalog-backend-module-unprocessed             0.4.0
  @backstage/plugin-catalog-backend                                1.18.0
  @backstage/plugin-catalog-common                                 1.0.22
  @backstage/plugin-catalog-graph                                  0.4.1
  @backstage/plugin-catalog-import                                 0.10.7
  @backstage/plugin-catalog-node                                   1.8.0
  @backstage/plugin-catalog-react                                  1.11.0
  @backstage/plugin-catalog-unprocessed-entities-common            0.0.1
  @backstage/plugin-catalog                                        1.18.0
  @backstage/plugin-events-backend-module-gitlab                   0.2.0
  @backstage/plugin-events-backend-test-utils                      0.1.24
  @backstage/plugin-events-backend                                 0.3.0
  @backstage/plugin-events-node                                    0.3.0
  @backstage/plugin-explore-common                                 0.0.2
  @backstage/plugin-github-actions                                 0.6.12
  @backstage/plugin-home-react                                     0.1.9
  @backstage/plugin-home                                           0.7.0
  @backstage/plugin-kubernetes-backend                             0.16.0
  @backstage/plugin-kubernetes-common                              0.7.5
  @backstage/plugin-kubernetes-node                                0.1.8
  @backstage/plugin-kubernetes-react                               0.3.1
  @backstage/plugin-kubernetes                                     0.11.6
  @backstage/plugin-org                                            0.6.21
  @backstage/plugin-permission-backend                             0.5.37
  @backstage/plugin-permission-common                              0.7.13
  @backstage/plugin-permission-node                                0.7.25
  @backstage/plugin-permission-react                               0.4.21
  @backstage/plugin-proxy-backend                                  0.4.12
  @backstage/plugin-scaffolder-backend-module-azure                0.1.6
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.4
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.4
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.4
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.6
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.4
  @backstage/plugin-scaffolder-backend-module-github               0.2.4
  @backstage/plugin-scaffolder-backend-module-gitlab               0.3.0
  @backstage/plugin-scaffolder-backend                             1.22.0
  @backstage/plugin-scaffolder-common                              1.5.1
  @backstage/plugin-scaffolder-node-test-utils                     0.1.0
  @backstage/plugin-scaffolder-node                                0.4.0
  @backstage/plugin-scaffolder-react                               1.8.1
  @backstage/plugin-scaffolder                                     1.19.0
  @backstage/plugin-search-backend-module-catalog                  0.1.18
  @backstage/plugin-search-backend-module-explore                  0.1.18
  @backstage/plugin-search-backend-module-pg                       0.5.23
  @backstage/plugin-search-backend-module-techdocs                 0.1.18
  @backstage/plugin-search-backend-node                            1.2.18
  @backstage/plugin-search-backend                                 1.5.4
  @backstage/plugin-search-common                                  1.2.11
  @backstage/plugin-search-react                                   1.7.7
  @backstage/plugin-search                                         1.4.7
  @backstage/plugin-techdocs-backend                               1.10.0
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.6
  @backstage/plugin-techdocs-node                                  1.12.0
  @backstage/plugin-techdocs-react                                 1.2.0
  @backstage/plugin-techdocs                                       1.10.1
  @backstage/plugin-user-settings                                  0.8.2
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.5.1
  @backstage/theme                                                 0.4.4, 0.5.2
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by vinzscam**
## Hey, I just made a Pull Request!

Fixes #23787

Passing down the `getInitiatorCredentials` method was causing the main issue

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts [hunk 1]
```diff
@@ -179,8 +179,8 @@ export class TaskManager implements TaskContext {
   }
 
   async getInitiatorCredentials(): Promise<BackstageCredentials> {
-    if (this.task.secrets && '__initiatorCredentials' in this.task.secrets) {
-      return JSON.parse(this.task.secrets.__initiatorCredentials);
+    if (this.task.secrets && this.task.secrets.initiatorCredentials) {
```
Comment (freben) [inline plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts:182]: We're these intentionally "hidden" before? Should they still be?

**Review by freben — APPROVED** [review]
Awesome, LGTM but @backstage/scaffolder-maintainers probably best have a look to

File: plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts (hunk 1 above)
Comment (vinzscam) [reply plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts:182]: my guess is that `__initiatorCredentials` was renamed to `initiatorCredentials` and this line was a left over, let's seeee

Comment (vinzscam) [reply plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts]: UPDATE: kept `__`

**Review by camilaibs — APPROVED** [review]

**Review by benjdlambert — APPROVED** [review]

**Comment by thanasiskoumpouras** [comment]
Hi. I have upgraded to 1.22.2 but I'm still getting the same error ```TypeError: Cannot read properties of undefined (reading 'secrets')```

I'm using the gitlab scaffolder ```    "@backstage/plugin-scaffolder-backend-module-gitlab": "^0.3.2",```

**Comment by vinzscam** [comment]
> Hi. I have upgraded to 1.22.2 but I'm still getting the same error `TypeError: Cannot read properties of undefined (reading 'secrets')`
> 
> I'm using the gitlab scaffolder ` "@backstage/plugin-scaffolder-backend-module-gitlab": "^0.3.2",`

mmm the fix was shipped in `1.25`

**Comment by saurabhssahu** [comment]
@vinzscam 
image

**Pull request merged**